src/drivers/audio/portaudio/portaudio_lib.c

```diff
@@ -221,6 +222,8 @@ PaError Pa_OpenStream(PaStream** stream,
 		return paInvalidDevice;
 	}
 
+	audio_dev->ad_ops->ad_ops_ioctl(audio_dev, ADIOCTL_SET_RATE, &rate);
```

Comment:
May be add ADIOCTL_GET_RATE and change a rate only if current rate differs from required?

Reply:
Thanks, I will add it.